app/src/main/java/org/mozilla/focus/telemetry/FirebaseEvent.java

```diff
@@ -43,11 +50,23 @@ private FirebaseEvent(@NonNull String category, @NonNull String method, @Nullabl
 
         // append null string will append 'null'
         // Android Studio also said we don't need to use a String builder here.
-        this.eventName = method + EVENT_NAME_SEPARATOR + object + EVENT_NAME_SEPARATOR + value;
-        if (!AppConstants.isReleaseBuild() && this.eventName.length() > MAX_LENGTH_EVENT_NAME) {
+        final String eventNamePrefix = method + EVENT_NAME_SEPARATOR + object + EVENT_NAME_SEPARATOR;
+        this.eventName = eventNamePrefix + value;
+        if (this.eventName.length() > MAX_LENGTH_EVENT_NAME) {
             // TODO: check eventName should start with[a-zA-Z] , contains only [a-zA-A0-9_], and shouldn't
             // TODO: start with ^(?!(firebase_|google_|ga_)).*
-            throw new IllegalArgumentException(this.eventName + " exceeds Firebase event name length limit " + this.eventName.length() + " of " + MAX_LENGTH_EVENT_NAME);
+            final int prefixLength = eventNamePrefix.length();
+
+            if (prefixLength > MAX_LENGTH_EVENT_NAME_PREFIX) {
+                handleError("Event[" + this.eventName + "]'s prefixLength too long  " + prefixLength + " of " + MAX_LENGTH_EVENT_NAME_PREFIX);
+            }
+            if (value != null && isValueInWhiteList(value)) {
+                int acceptValueLength = MAX_LENGTH_EVENT_NAME - eventNamePrefix.length();
```

Comment:
Maybe we would like to have a test to make sure this is handled correctly? probably through initializing a whitelisted event and check if the length is 40 after the initialization.

Reply:
Do you mean I should create a share preference key <-> firebase event change_setting_pref_key mapping? I think it'll need hardcoded strings and will be hard to maintain. So here I just whitelist existing events and trim them automatically. How do you think?

Reply:
Oh I didn't make it clear sorry. I mean adding a test.